api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/DispensingDatasetDefinitionEvaluator.java

```diff
@@ -63,70 +65,71 @@ public DataSet evaluate(DataSetDefinition dataSetDefinition, EvaluationContext c
+            }
 
-            Set<Integer> orderIds = Stream.of(drugOrderGroupIds.split(","))
-                    .map(Integer::parseInt)
-                    .collect(Collectors.toSet());
 
+            Connection connection = null;
+            try {
+                connection = sqlConnection();
+                String allPatients = Joiner.on(",").join(results.stream().map(Object -> Object[1]).collect(Collectors.toList()));
+                String drugOrderGroupIds = Joiner.on(",").join(results.stream().map(Object -> Object[0]).collect(Collectors.toList()));
 
-            Map<Integer, List<PersonDemographics>> demographics = getPatientDemographics(connection, allPatients);
+                Set<Integer> orderIds = Stream.of(drugOrderGroupIds.split(","))
+                        .map(Integer::parseInt)
+                        .collect(Collectors.toSet());
 
-            PatientDataHelper pdh = new PatientDataHelper();
-            for (int i :orderIds) {
-                DataSetRow row = new DataSetRow();
 
-                List<Object[]> drugOrder = results.stream().filter(Object -> Object[0].equals(i)).collect(Collectors.toList());
-                for(Object[] o :drugOrder) {
-                    List<PersonDemographics> personDemographics = demographics.get(o[1]);
-                    PersonDemographics personDemos = personDemographics != null && personDemographics.size() > 0 ? personDemographics.get(0) : new PersonDemographics();
+                Map<Integer, List<PersonDemographics>> demographics = getPatientDemographics(connection, allPatients);
 
-                    int question = (int)o[2];
-                    String drug = String.valueOf(o[4]);
+                PatientDataHelper pdh = new PatientDataHelper();
+                for (int i : orderIds) {
+                    DataSetRow row = new DataSetRow();
 
-                    if(question==1282){
-                        pdh.addCol(row, "drug", drugNames.get((int)o[4]));
-                    }
-                    if(question==1443){
-                        pdh.addCol(row, "quantity", o[3]);
-                    }
-                    pdh.addCol(row, "date", DateUtil.formatDate(DateUtil.parseYmd(String.valueOf(o[5])), "yyyy-MM-dd"));
-                    pdh.addCol(row, "Patient_ID", processString(personDemos.getIdentifiers()).get("e1731641-30ab-102d-86b0-7a5022ba4115"));
+                    List<Object[]> drugOrder = results.stream().filter(Object -> Object[0].equals(i)).collect(Collectors.toList());
+                    for (Object[] o : drugOrder) {
+                        List<PersonDemographics> personDemographics = demographics.get(o[1]);
+                        PersonDemographics personDemos = personDemographics != null && personDemographics.size() > 0 ? personDemographics.get(0) : new PersonDemographics();
 
+                        int question = (int) o[2];
+                        String drug = String.valueOf(o[4]);
```

Comment:
@dbaluku the statement `String drug = String.valueOf(o[4]);` looks like it is not used any where, please remove it, if its not used

Reply:
Done